GSoC-2025/movement: support for Kalman filters (Angkul).md

```diff
@@ -0,0 +1,94 @@
+
+## Project Proposal
+
+- **Synopsis**
+
+    This project focuses on implementing the Kalman Filter in movement for data cleaning and filtering, specifically to smooth position, velocity, and acceleration data in animal tracking data. This helps in reducing noise in data and improve the data consistency and quality. Additionally, the project tackles the identity-switch problem in multi-animal tracking, ensuring more reliable tracking and analysis. 
+    This project is important because this aims to reduce the noisiness in animal tracking data and thus improving the data quailty for analysing motion tracking. 
+    
+    Deliverables include the Kalman Filter implementation, extending Kalman Filter to fix identity switch, testing-documentation of the new functionality and an example in movement gallery.  With the help of this Kalman Filter implementation, the open-source community will gain cleaner, more consistent, and high-quality data for motion tracking analysis benefiting the researchers, scientists.
+
+- **Implementation timeline**
+    
+    **Deliverables**
+    - Kalman Filter for Smoothing of position, velocity, acceleration
+    - Tests for the Kalman Filter Smoothing
+    - Documentation for Kalman Filter function and smoothing
+    - Example use case of Kalman Filter in movement gallery
+
+    **Stretch Goals**
+    - Kalman Filter for Identity Switch correction in Multi-Animal Tracking
+    - Extended tests and documentation for Identity correction
+
+    **Weekly Timeline**
+
+    | Time Frame | Tasks & Deliverables |
+    |------------|---------------------|
+    | **Pre GSOC period**  <br> March 24 - May 8 | - Take a deeper dive into the work done so far, learn about the Kalman filter, explore the other existing implementations of the Kalman Filter.  <br> - Explore the movement codebase, docs and testing things, fix bugs, implement examples to understand the internal working of movement functions. |
+    | **Community Bonding Period (Before week 1)**  <br> May 8 - June 1 | - Much time won’t be wasted as I’m aware of most parts of the codebase and know the mentors well and the development environment is already set.  <br> - Discuss and decide the best possible method to implement the Kalman Filter with the mentor. |
+    | **Week 1**  <br> June 2 - June 9 | - Implement the basic Kalman Filter for position smoothing.  <br> - Test the position smoothing feature on the movement dataset. |
+    | **Week 2**  <br> June 10 - June 17 | - Extend the Kalman Filter to handle velocity and acceleration smoothing.  <br> - Derive the velocity and acceleration data from the position data using the kinematics module for testing purposes.  <br> - Test the velocity and acceleration smoothing on the derived dataset. |
+    | **Week 3**  <br> June 18 - June 25 | - Discuss the current progress with the mentor and ask for review and suggestions.  <br> - Work on the suggestions and unit testing for the smoothing functionality. |
```

Comment:
One suggestion here: I'd prefer to review current progress immediately after the position-only implementation, and before you proceed to adding other variables as well.

Reply:
In general we prefer to work with short PRs and short feedback loops.